docs/vnext/contract.md

```diff
@@ -35,9 +35,9 @@ For details about resource APIs, Refer to the OpenAPI 3.0 based resource API [sp
 
 ## Long-running Operation Reference
 
-The Extensibility Host supports two types of long-running operations: resource based long-running operations (RELO) and stepwise long-running operations (LRO). Comprehensive details regarding each of these operation types can be found in the [Microsoft REST API Guideline](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#13-long-running-operations). Extensibility providers should align any other long-running operation patterns implemented by the underlying API with these two classifications.
+The Extensibility Host supports two types of long-running operations: resource based long-running operations (RELO) and stepwise long-running operations (LRO). Comprehensive details regarding each of these operation types can be found in the [Microsoft REST API Guideline](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#13-long-running-operations). Extensions should align any other long-running operation patterns implemented by the underlying API with these two classifications.
 
-> As highlighted in [Microsoft REST API Guideline](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#1441-responsiveness), it is recommended that invocations taking longer than 1 second to respond in the 99th percentile SHOULD employ the long-running operation pattern. However, this guideline is not mandatory for extensibility providers, as it might not perfectly align with the nuances of the underlying API implementation. Nonetheless, extensibility providers are strongly encouraged to enhance their overall system responsiveness. For instance, a slow synchronous API can be converted into an asynchronous LRO counterpart. This can be achieved by exposing an operation resource dedicated to the Extensibility Host, facilitating real-time tracking of operation status and progress. In this case, the operation resource is managed by the extensibility provider, leaving the underlying API unchanged.
+> As highlighted in [Microsoft REST API Guideline](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#1441-responsiveness), it is recommended that invocations taking longer than 1 second to respond in the 99th percentile SHOULD employ the long-running operation pattern. However, this guideline is not mandatory for extensions, as it might not perfectly align with the nuances of the underlying API implementation. Nonetheless, extensions are strongly encouraged to enhance their overall system responsiveness. For instance, a slow synchronous API can be converted into an asynchronous LRO counterpart. This can be achieved by exposing an operation resource dedicated to the Extensibility Host, facilitating real-time tracking of operation status and progress. In this case, the operation resource is managed by the extension, leaving the underlying API unchanged.

```

Comment:
> Is it worth supporting both synchronous and asynchronous operations from extensions?

I think we have to support both to accommodate synchronous and asynchronous MS Graph workloads.

> Nonetheless, extensions are strongly encouraged to enhance their overall system responsiveness. For instance, a slow synchronous API can be converted into an asynchronous LRO counterpart. This can be achieved by exposing an operation resource dedicated to the Extensibility Host, facilitating real-time tracking of operation status and progress. In this case, the operation resource is managed by the extension, leaving the underlying API unchanged.

> I remember from a previous design that it was a goal to have the extension remain stateless. Is that still the case? If that's still true how will asynchronous operations work?

Implementing LRO support in an extension for synchronous operations requires tracking the status of operations using some kind of caching or storage system inside the extension, making the whole extension stateful (the web service can remain stateless). The strategy can serve as an optional enhancement for extensions facing slow synchronous underlying APIs to improve responsiveness. That being said, considering our current extensions and upcoming ev2 extension may not need this enhancement, and given the deprecation of the MS REST API guideline, it might be justified to remove this section from the doc.

Reply:
> I think we have to support both to accommodate synchronous and asynchronous MS Graph workloads and the EV2 APIs.

I guess what I was asking is whether there's a need to support synchronous operations at all, given that you have to support asynchronous.

Reply:
Ah I see. I think there's still a need for supporting synchronous operations because the majority of the MS graph APIs are synchronous. However, technically speaking, there is no difference between a synchronous operation response that has a null status and an immediate asynchronous RELO operation response with status "Succeeded", and both scenarios are handled by the same code within the deployment job, so it does seem that we are essentially "only" supporting asynchronous operations 🤔.